clients/drcachesim/CMakeLists.txt

```diff
@@ -171,6 +171,7 @@ target_link_libraries(drmemtrace_invariant_checker drdecode)
 
```

Comment:
Please update the PR description:

> Fixes the corner case of the wrong previous instruction by using last_instr_in_cur_context_ instead of prev_instr_ when dealing with kernel xfers.

I don't see this change.  This line should be removed I assume.

> Adds unit tests to check for PC discontinuities in the transitions between instructions and kernel xfer markers
> Adds caching around the decoding, to prevent repeated calls to the expensive decode_from_copy() function.

These 2 are not changes "to how the invariant checker checks for discontinuities between kernel xfer markers": please rearrange.

Probably worth mentioning that the big redundant pre_signal_whatever_it_was check is removed.

> Issue: https://github.com/DynamoRIO/dynamorio/issues/5912

Does this fix that issue?